Source/Dafny/Printer.cs

```diff
@@ -1495,8 +1495,13 @@ private bool PrintModeSkipGeneral(Bpl.IToken tok, string fileBeingPrinted)
           wr.Write("assume ");
         }
         PrintExpression(update.Expr, true);
+      } else if (s is AssignOrReturnStmt) {
+        var stmt = (AssignOrReturnStmt)s;
+        wr.Write(" :- ");
```

Comment:
Somewhere, the Printer should also print the new `:- M(...);` statement (that is, the one with no LHS).

Reply:
Also, if `/rprint` is being used, make the printing also spill out the desugaring. There is some precedence for this (the auto-generated members of `iterator` come to mind, but I think there may also be statement forms that do this).

Reply:
The no-LHS `:-` statement is not supported yet in the commits you reviewed, but I'll make sure to also add printing for it.
I'm adding printing of the desugared code inside comments like this:

```
method CheckThatVerifierRunsInsideDesugaredExprs(r1: NatOutcome, r2: NatOutcome) returns (res: NatOutcome)
  decreases r1, r2
{
  var a: nat :- MakeNatSuccess(assert r1 == r2; 12)
  /*---------- desugared ----------
  var valueOrError0: NatOutcome := MakeNatSuccess(assert r1 == r2; 12);
  if valueOrError0.IsFailure() {
    return valueOrError0.PropagateFailure();
  }
  a := valueOrError0.Extract();
  ---------- end desugared ----------*/;
  res := MakeNatSuccess(a);
}
```